devito/stencilkernel.py

```diff
@@ -97,7 +98,8 @@ def __init__(self, stencils, **kwargs):
 
         # Introduce C-level profiling infrastructure
         self.sections = OrderedDict()
-        nodes = self._profile_sections(nodes)
+        if self.profiler:
```

Comment:
@mlange05 @FabioLuporini 
Again, profiling/bug is not the point of this PR. Can we focus on this and make another one for debug if there is a better way to do it.